app/lib/katello/util/path_with_substitutions.rb

```diff
@@ -12,12 +14,23 @@ def initialize(path, substitutions)
         @substitutions = substitutions
         @path = path
         @resolved = []
+        generate_substitutions_from_path if @substitutions.empty?
+      end
+
+      def split_path
+        @split ||= @path.split('/').reject(&:blank?)
       end
 
+      def generate_substitutions_from_path
+        arches = split_path & ARCHITECTURES
+        arch = arches.first
```

Comment:
should this raise an exception if arches is bigger than 1?

Reply:
I'm not sure.. that would mean the repo would have a path like `/content/dist/rhel8/8/x86_64/product/x86_64` so to generate an incorrect substitution, it would have to have two arches in its path and one be incorrect, i.e. `/content/dist/rhel8/8/aarch64/product/x86_64`

One thing I was thinking of adding is a check in the fifth position for the arch first since that seems to be the agreed upon convention for RHEL8.

That would look like this:
- Check 5th position of the path for arch in ARCHITECTURES, if exists, use that in substitutions and skip checking the full path
- Check full path for arch in ARCHICTECTURES, if any exist, use the first one found in substitutions.
- If no arches found, do nothing.

This would at least check the expected place for the arch first before checking the whole path